packages/teleterm/src/ui/TabHost/TabHost.test.tsx

```diff
@@ -0,0 +1,147 @@
+import { fireEvent, render } from 'design/utils/testing';
+import React from 'react';
+import { TabHost } from 'teleterm/ui/TabHost/TabHost';
+import { MockAppContextProvider } from 'teleterm/ui/fixtures/MockAppContextProvider';
+import { Document, DocumentsService } from 'teleterm/ui/services/docs';
+import { KeyboardShortcutsService } from 'teleterm/ui/services/keyboardShortcuts';
+import {
+  MainProcessClient,
+  TabContextMenuOptions,
+} from 'teleterm/mainProcess/types';
+import { getMockDocuments } from 'teleterm/ui/fixtures/mocks';
+import { ClustersService } from 'teleterm/ui/services/clusters';
+
+function getTestSetup() {
```

Comment:
Something is causing TabHost.test.tsx to output a lot of warning related to `DocumentTerminal`:

<details>
<summary>DocumentTerminal warning</summary>

```
  console.error
    Warning: An update to DocumentTerminal inside a test was not wrapped in act(...).
    
    When testing, code that causes React state updates should be wrapped into act(...):
    
    act(() => {
      /* fire events that update state */
    });
    /* assert on the output */
    
    This ensures that you're testing the behavior the user would see in the browser. Learn more at https://fb.me/react-wrap-tests-with-act
        in DocumentTerminal (created by DocumentTerminalContainer)
        in DocumentTerminalContainer (created by MemoizedDocument)
        in MemoizedDocument (created by TabHost)
        in div (created by styled.div)
        in styled.div (created by TabHost)
        in TabHost
        in AppContextProvider (created by MockAppContextProvider)
        in DndProvider (created by MockAppContextProvider)
        in MockAppContextProvider
        in StyleSheetManager (created by ThemeProvider)
        in ThemeProvider (created by ThemeProvider)
        in ThemeProvider (created by Providers)
        in Providers

      43 |       return [data, null] as [R, Error];
      44 |     } catch (err) {
    > 45 |       setState({
         |       ^
      46 |         ...state,
      47 |         status: 'error',
      48 |         statusText: err.message,

      at printWarning (node_modules/react-dom/cjs/react-dom.development.js:88:30)
      at error (node_modules/react-dom/cjs/react-dom.development.js:60:5)
      at warnIfNotCurrentlyActingUpdatesInDEV (node_modules/react-dom/cjs/react-dom.development.js:23284:7)
      at setState (node_modules/react-dom/cjs/react-dom.development.js:15656:9)
      at run (packages/teleterm/src/ui/useAsync.ts:45:7)
```
</details>

I didn't have much time to dive into this, but do you know what could cause this?

Reply:
Right, they are caused by running async actions in `DocumentTerminal` (there are also errors about missing canvas API in test env). I changed the documents mock so it won't include terminal documents, testing of each document type will be done separately.

Reply:
This makes me wonder if we should have a global fixture with documents like we do in `teleterm/ui/fixtures/mocks`. As far as I understand, here below is another tests which does something extraordinary because of how the fixtures are set up. I left another comment under it.

https://github.com/gravitational/webapps/blob/72a31a8c51b5fd5c14efb1f95f808819d788bdc2/packages/teleterm/src/ui/TabHost/TabHost.test.tsx#L147

I'm a fan of crafting the setup for each test by hand, even if that means a lot of boilerplate code. Though I understand that the setup for TabHost tests is quite complicated and we need to mock a lot of stuff. Maybe adding `mockedDocuments` as an arg to `getTestSetup` would allow us to set up each test so that it includes only what's absolutely necessary? If "should open context menu" operates on only a single document, it could have only that document in its mocked documents, for example.

Though again, I'm not bent on having it this way, just trying to understand if this is something that we usually do and if it's helping us in certain scenarios.

Reply:
I've tried to address your comment, global `getMockedDocuments()` is removed, I added local functions to create mocks, and whenever possible I'm taking from them only what is necessary.